libbluray needs to build .jar for bluray menus to work

<!-- Don't request update of package. We have a script for that. https://alpha.de.repo.voidlinux.org/void-updates/void-updates.txt . However, a quality pull request may help. -->
### System

* xuname:  
  Void 5.4.38_1 x86_64 GenuineIntel uptodate rrrmFFFF
* package:  
  libbluray-1.2.0_1

### Expected behavior
Blurays play fine in VLC, including their menus.

### Actual behavior
VLC tells me that it can only play blurays in no-menus mode because I don't have Java installed, even though I do have Java installed.

### But what does this have to do with the libbluray package?
To make a long story short, I did some digging and figured out that the problem is that I don't have libbluray.jar anywhere on my system (and that multiple developers deserve to suffer a public drubbing for their lazy error messages.)
The xbps template for the libbluray package has a flag set to explicitly not build the .jar. I don't know why that option is set, but the .jar seems to be important for anyone who wants to watch blurays with menus. (For TV shows, this includes having any ability to play the episodes in their correct order.)

For anyone looking to confirm the problem, here's the VLC terminal output I get:

```VLC media player 3.0.10 Vetinari (revision 3.0.10-0-g7f145afa84)
[00005592364e7660] main libvlc: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-####'
[0000559236576530] main playlist: playlist is empty
syntax error: line 1
bad entry at or around line 1
aacs.c:546: Error calculating media key. Missing right processing key ?
bdj.c:569: libbluray-j2se-1.2.0.jar not found.
bdj.c:756:** BD-J check: Failed to load libbluray.jar
bdj.c:569: libbluray-j2se-1.2.0.jar not found.
bdj.c:756: BD-J check: Failed to load libbluray.jar
[00007f9ac8001120] libbluray demux: First play: 0, Top menu: 0
HDMV Titles: 0, BD-J Titles: 3, Other: 3
[00007f9ac8001120] libbluray demux error: BD-J menus not supported. Playing without menus. BD-J support: 1, JVM found: 1, JVM usable: 0
[00007f9ac8001120] libbluray demux: Adding ES 4113 select 1
[00007f9ac8001120] libbluray demux: Adding ES 4352 select 1
[00005592365b09c0] main audio output error: too low audio sample frequency (0)
[00007f9acc0e6500] main decoder error: failed to create audio output
[00007f9ac8001120] libbluray demux: Adding ES 4353 select 0
[00007f9ac8001120] libbluray demux: Adding ES 4608 select 0
[00005592365b09c0] vlcpulse audio output error: digital pass-through stream connection failure: Not supported
[00005592365b09c0] main audio output error: module not functional
[00007f9acc0e6500] main decoder error: failed to create audio output
libva info: VA-API version 1.7.0
libva info: Trying to open /usr/lib/dri/radeonsi_drv_video.so
libva info: Found init function __vaDriverInit_1_7
libva info: va_openDriver() returns 0
mesa: for the   --simplifycfg-sink-common option: may only occur zero or one times!
mesa: for the   --global-isel-abort option: may only occur zero or one times!
[00007f9ab8005560] glconv_vaapi_x11 gl error: vaDeriveImage: operation failed
[00007f9ab4050fe0] main video output error: video output creation failed
[00007f9acc004b40] main decoder error: failed to create video output
mesa: for the   --simplifycfg-sink-common option: may only occur zero or one times!
mesa: for the   --global-isel-abort option: may only occur zero or one times!
[00007f9acc004b40] avcodec decoder: Using G3DVL VDPAU Driver Shared Library version 1.0 for hardware decoding
[00007f9acc004b40] main decoder error: buffer deadlock prevented
[00007f9ac8001120] libbluray demux: Closing overlays.
^CQObject::~QObject: Timers cannot be stopped from another thread
```

Aside from the messages about libbluray.jar not being found, this is the one that made me stop assuming my JAVA_HOME path was broken somehow:
`[00007f9ac8001120] libbluray demux error: BD-J menus not supported. Playing without menus. BD-J support: 1, JVM found: 1, JVM usable: 0`
Digging into the libbluray source code revealed that "JVM usable: 0" also means that libbluray.jar is missing, not that there's anything actually wrong with my JVM.

--- a/srcpkgs/libbluray/template
+++ b/srcpkgs/libbluray/template
@@ -1,10 +1,10 @@
 # Template file for 'libbluray'
 pkgname=libbluray
 version=1.2.0
-revision=1
+revision=2
 build_style=gnu-configure
-configure_args="--disable-static --disable-optimizations --disable-bdjava-jar"
-hostmakedepends="pkg-config"
+configure_args="--disable-static --disable-optimizations"
+hostmakedepends="pkg-config apache-ant openjdk8"
 makedepends="libxml2-devel fontconfig-devel"
 short_desc="Library to access Blu-Ray disks for video playback"
 maintainer="Orphaned <orphan@voidlinux.org>"
@@ -12,6 +12,14 @@ license="LGPL-2.1-or-later"
 homepage="https://www.videolan.org/developers/libbluray.html"
 distfiles="https://download.videolan.org/pub/videolan/libbluray/${version}/libbluray-${version}.tar.bz2"
 checksum=cd41ea06fd2512a77ebf63872873641908ef81ce2fe4e4c842f6035a47696c11
+
+pre_build() {
+	export JAVA_HOME="/usr/lib/jvm/java-1.8-openjdk"
+}
+
+pre_install() {
+	export JAVA_HOME="/usr/lib/jvm/java-1.8-openjdk"
+}
 
 libbluray-devel_package() {
 	depends="${makedepends} ${sourcepkg}>=${version}_${revision}"

